src/target/ext/register_info_override.rs

```diff
@@ -0,0 +1,59 @@
+//! (LLDB extension) Override the register info specified by `Target::Arch`.
+use crate::arch::lldb::Register;
+use crate::target::Target;
+
+/// (LLDB extension) If your target is using the `qRegisterInfo` packet, this
+/// token hast to be returned from the `register_info` function to guarantee
+/// that the callback function to write the register info has been invoked.
+pub struct CallbackToken<'a>(pub(crate) core::marker::PhantomData<&'a *mut ()>);
+
+/// (LLDB extension) This struct is used internally by `gdbstub` to wrap a
+/// callback function which has to be used to direct the `qRegisterInfo` query.
+pub struct Callback<'a> {
+    /// The callback function that is directing the `qRegisterInfo` query.
+    pub(crate) cb: &'a mut dyn FnMut(Option<Register<'_>>),
+    /// A token to guarantee the callback has been used.
+    pub(crate) token: CallbackToken<'a>,
+}
+
+/// (LLDB extension) Trait which acts as a forwarder for the callback. Use the
+/// `write` method to write register info of a single register to the LLDB
+/// client or the `done` method to end the `qRegisterInfo` query. Both methods
+/// will return a `CallbackToken`, which has to be returned from the
+/// `register_info` implementation.
+impl<'a> Callback<'a> {
+    /// The `qRegisterInfo` query shall be concluded.
+    #[inline(always)]
+    pub fn done(self) -> CallbackToken<'a> {
+        (self.cb)(None);
+        self.token
+    }
+
+    /// Write the register info of a single register.
+    #[inline(always)]
+    pub fn write(self, reg: Register<'_>) -> CallbackToken<'a> {
+        (self.cb)(Some(reg));
+        self.token
+    }
+}
+
+/// (LLDB extension) Target Extension - Override the target register info
+/// specified by `Target::Arch`.
+///
+/// _Note:_ Unless you're working with a particularly dynamic,
+/// runtime-configurable target, it's unlikely that you'll need to implement
+/// this extension.
+pub trait RegisterInfoOverride: Target {
+    /// Invoke `reg_info.write(reg)` where `reg` is a [Register
+    /// struct](../../../arch/lldb/struct.Register.html) to write information of
```

Comment:
nit: please use intradoc links rahter than linking to HTML directly: https://rust-lang.github.io/rfcs/1946-intra-rustdoc-links.html